plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ColumnWithAliasFormatterLegacy.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.spi.connector.ConnectorSession;
+
+public class ColumnWithAliasFormatterLegacy
+        implements ColumnWithAliasFormatter
+{
+    @Override
+    public JdbcColumnHandle format(JdbcColumnHandle column, int nextSyntheticColumnId, ConnectorSession session)
```

Comment:
Instead two distinct implementations let's have one with the signature `format(ConnectorSession session, JdbcColumnHandle column, OptionalInt maximumLength, int nextColumnId)`.

if maximumLength is empty return legacy value, otherwise: apply logic as in the new implementation.